### PR TITLE
FM2-420: Integration tests for type level $everything operation on Patients

### PR DESCRIPTION
**Pull request by Medhavi-16**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-420

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by Medhavi-16** [comment]
@ibacher @varung-31 PTAL. Currently, the validation of what resources should be there is not added. There were 41 resources and I was not sure if creating variables for 41 resource ids would be the correct way to go. Would you please give your thoughts on this?

File: integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/PatientFhirResourceProviderIntegrationTest.java [hunk 1]
```diff
@@ -659,7 +745,22 @@ public void shouldReturnForPatientEverythingWhenCountIsSpecifiedAsXml() throws E
...
+		
 		return validResources;
 	}
+
+	private Set<ResourceType> getEverythingValidResourceTypes() {
+		Set<ResourceType> validTypes = new HashSet<>();
+
+		validTypes.add(ResourceType.Patient);
+		validTypes.add(ResourceType.Observation);
+		validTypes.add(ResourceType.Procedure);
```
Comment (varung-31) [inline integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/PatientFhirResourceProviderIntegrationTest.java:757]: We don't use `Procedure` in the _revinclude implementation for Patient resource.

**Review by varung-31 — CHANGES_REQUESTED** [review]
Thanks for the work on this @Medhavi-16 ! Just a minor comment from me.

**Review by ibacher — APPROVED** [review]

**Comment by ibacher** [comment]
Thanks @Medhavi-16!

**Pull request merged**